docs/components/saas/byok/aws-kms-setup.md

```diff
@@ -0,0 +1,178 @@
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::<customer-aws-account>:root"
+      },
+      "Action": "kms:*",
+      "Resource": "*"
+    },
+    {
+      "Sid": "Allow Camunda tenant IAM Role basic key access",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "<tenant-role-arn>"
+      },
+      "Action": [
+        "kms:Encrypt",
+        "kms:Decrypt",
+        "kms:ReEncrypt*",
+        "kms:DescribeKey",
+        "kms:GenerateDataKey*"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Sid": "Allow Camunda tenant IAM Role to create grants for provisioning encrypted EBS volumes",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "<tenant-role-arn>"
+      },
+      "Action": ["kms:CreateGrant", "kms:ListGrants", "kms:RevokeGrant"],
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+:::warning Key policy guidance
+
+- Ensure you do not restrict the Camunda cluster Amazon Role from required KMS actions; otherwise, encryption will fail.
+- Key rotation must be managed in AWS KMS; Camunda cannot rotate your keys.
+- Any revocation of access immediately breaks the Camunda cluster.
+  :::
+
+7. Click **Finish** to create the key.
+8. Copy the **KMS Key ARN**; you will need it in the Camunda Console.
+
+:::note
+[Insert screenshot of Amazon KMS key details showing Key ARN]
+:::
+
+## Step 3: Associate KMS key with Camunda cluster
+
+1. Return to the Camunda Console.
+2. On the cluster details page, locate the **KMS Key ARN** input field.
+   - For dual region, two fields will be available—enter the correct key for each region.
+3. Paste your Amazon KMS Key ARN(s) from Step 2.
+4. Confirm and apply the key. The Camunda Operator provisions storage using your key for:
+   - Document handling storage
+   - Backup storage
+   - Orchestration cluster persistent disks
+   - Elasticsearch persistent disks
+
+:::note
+Once a key is applied, it cannot be edited or replaced—even if the key was invalid or encryption failed. To change keys, you must create a new cluster.
+:::
+
+:::note
+[Insert screenshot showing KMS Key ARN input in cluster details]
+:::
+
+## Step 4: Verify encryption and logging
+
+After associating your Amazon KMS key with the Camunda cluster, verify that encryption is working as expected.
+
+### Verify encryption
+
+- In the Camunda Console, check the cluster details page to confirm that the **KMS Key ARN** is correctly applied.
+- Verify that all cluster storage volumes are encrypted with the configured key:
+  - Document handling storage
+  - Backup storage
+  - Orchestration cluster persistent disks
+  - Elasticsearch persistent disks
+- If needed, check the encryption status directly in the underlying storage service (e.g., Amazon EBS, S3, or Elasticsearch).
+
+### Monitor KMS usage
+
+- **Amazon CloudTrail** automatically logs all KMS operations, including key usage (`Encrypt`, `Decrypt`, `GenerateDataKey`).
+- Use **Amazon CloudWatch** to create alarms for:
```

Comment:
Could we expand to explain that alarms should be set up in a customer's AWs account, their KMS service.